src/averageAges.js

```diff
@@ -56,6 +81,19 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const kids = people.filter(kid =>
+    onlyWithSon
+      ? people.some(person => kid.mother === person.name) && male(kid)
+      : people.some(person => kid.mother === person.name)
+  );
```

Comment:
could be bit shorter
```suggestion
  const kids = people.filter(kid =>
    people.some(person => kid.mother === person.name)
    && (onlyWithSon ? male(kid) : true)
  );
```